modules/tinymce/src/themes/silver/test/ts/webdriver/toolbar/SearchableMenuTypingTest.ts

```diff
@@ -0,0 +1,366 @@
+        Attribute.remove(activeElem, 'data-sendkeys');
+      }
+    );
+  };
+
+  const onActiveElement = <A>(doc: SugarElement<Document>, f: (elem: SugarElement<any>) => A): A => {
+    return Focus.active(doc).fold<A>(
+      () => {
+        throw new Error('Could not find focused element');
+      },
+      (focused) => f(focused)
+    );
+  };
+
+  const pSendKeysToActiveInput = (_doc: SugarElement<Document>, keys: KeyPressAdt[]) => {
+    return pSendToActiveElement(hook.root(), async (_elem, selector) =>
+      await RealKeys.pSendKeysOn(selector, keys)
+    );
+  };
+
+  // We are currying this so that it works more easily with onActiveElement
+  const assertTextState = (expected: { start: number; end: number; text: string }) => (input: SugarElement<HTMLInputElement>) => {
+    Assertions.assertEq(
+      'Checking input state (selection positions and value)',
+      expected,
+      {
+        start: input.dom.selectionStart,
+        end: input.dom.selectionEnd,
+        text: Value.get(input)
+      }
+    );
+  };
+
+  // We are currying this so that it works more easily with onActiveElement
+  const assertCursorAtEndOfText = (expectedText: string) => (input: SugarElement<HTMLInputElement>) => {
+    Assertions.assertEq(
+      'Checking input state (selection positions and value)',
+      { start: expectedText.length, end: expectedText.length, text: expectedText },
+      {
+        start: input.dom.selectionStart,
+        end: input.dom.selectionEnd,
+        text: Value.get(input)
+      }
+    );
+  };
+
+  const pAssertTieredMenuStructure = async (label: string, sink: SugarElement<HTMLElement>, children: StructAssert[]) => {
+    const tmenu = await UiFinder.pWaitForVisible(
+      `Waiting for TieredMenu [${label}]`,
+      sink,
+      '.tox-tiered-menu:has(.tox-selected-menu)'
+    );
+
+    Assertions.assertStructure(
+      `Asserting TieredMenu structure [${label}]`,
+      ApproxStructure.build((s, _str, _arr) => {
+        return s.element('div', {
+          children
+        });
+      }),
+      tmenu
+    );
+  };
+
+  const hook = TestHelpers.GuiSetup.bddSetup(
+    (store, _doc, _body) => GuiFactory.build(
+      renderMenuButton(
+        {
+          text: Optional.some('MailMerge'),
+          icon: Optional.none(),
+          tooltip: Optional.none(),
+          onSetup: Fun.constant(Fun.noop),
+          search: Optional.some({ placeholder: Optional.none() }),
+          fetch: fetchMailMergeData({
+            // To test that <left> and <right> etc. are not being
+            // processed by the menu, we need to have menus that can expand / collapse
+            // even when there is a search term.
+            collapseSearchResults: false
+          }, store)
+        },
+        'prefix',
+        helpers.backstage(),
+        Optional.none()
+      )
+    )
+  );
+
+  // The <space> key appears to be missing from KeyEffects in bedrock
+  const spaceKey = RealKeys.text('\uE00D');
```

Comment:
Could we move this into RealKeys so that we can fix that in one spot?

Reply:
Or log a ticket to add it to bedrock?

Reply:
Yeah, I think logging a ticket is a better option. It would be weird to have some in RealKeys and some in bedrock.

- [ ] log a ticket to add `space` to bedrock.

Reply:
Updated: logged a ticket and referenced it.